src/twisted/conch/test/test_conch.py

```diff
@@ -32,28 +32,27 @@
 except ImportError:
     pass
 
-try:
-    import pyasn1
-except ImportError:
-    pyasn1 = None
-
+pyasn1 = requireModule('pyasn1')
 cryptography = requireModule("cryptography")
+
 if cryptography:
     from twisted.conch.avatar import ConchUser
     from twisted.conch.ssh.session import ISession, SSHSession, wrapProtocol
 else:
     from twisted.conch.interfaces import ISession
 
-    class ConchUser:
+    class ConchUser:  # type: ignore[no-redef]
         pass
 try:
     from twisted.conch.scripts.conch import (
-        SSHSession as StdioInteractingSession
+        SSHSession as _StdioInteractingSession
```

Comment:
Can't use `no-redef`? This is less desirable than a comment, since a fix to `mypy` to handle these import cases. won't notice that this change isn't needed any more.

Reply:
OK I think this is here for the line where assign `None`.

I would suggest ` # type: ignore[assignment]` on the line where we assign `None` instead.